### PR TITLE
Test ehnancement, integrate Travis build

### PR DESCRIPTION
**Pull request by peter279k**
Reopen PR #13.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     },
     "require-dev": {
         "ext-xdebug": "*",
-        "phpunit/phpunit": "^5.4",
+        "phpunit/phpunit": "^5.4|^6.5",
```
Comment (Lewiscowles1986) [inline composer.json:11]: Was there a feature that needed 6.5?

If there are reasons please raise an issue and then link the issue in the commit or PR

File: composer.json [hunk 2]
```diff
@@ -24,5 +24,10 @@
         "psr-4": {
             "lewiscowles\\Rfc\\": "src"
         }
+    },
+    "autoload-dev": {
+        "psr-4": {
+            "lewiscowles\\Rfc\\": "tests"
```
Comment (Lewiscowles1986) [inline composer.json:30]: As in other repo tests really need their own namespace, they are not part of the body of code.

File: src/Attachment.php [hunk 3]
```diff
@@ -7,7 +7,7 @@
 use Psr\Http\Message\StreamInterface;
 
 
-Final class Attachment extends AbstractNode {
+final class Attachment extends AbstractNode {
```
Comment (Lewiscowles1986) [inline src/Attachment.php:10]: specificity over caps... This really should come with a coding style issue and PR first

File: src/Envelope.php [hunk 4]
```diff
@@ -5,7 +5,7 @@
 use lewiscowles\Rfc\NodeInterface;
 
 
-Final class Envelope implements NodeInterface {
+final class Envelope implements NodeInterface {
```
Comment (Lewiscowles1986) [inline src/Envelope.php:8]: specificity over caps... This really should come with a coding style issue and PR first

File: src/FormBody.php [hunk 5]
```diff
@@ -9,7 +9,7 @@
 use Psr\Http\Message\StreamInterface;
 
 
-Final class FormBody {
+final class FormBody {
```
Comment (Lewiscowles1986) [inline src/FormBody.php:12]: specificity over caps... This really should come with a coding style issue and PR first

File: src/FormInput.php [hunk 6]
```diff
@@ -5,7 +5,7 @@
 use lewiscowles\Rfc\AbstractNode;
 
 
-Final class FormInput extends AbstractNode {
+final class FormInput extends AbstractNode {
```
Comment (Lewiscowles1986) [inline src/FormInput.php:8]: specificity over caps... This really should come with a coding style issue and PR first

File: tests/BasicTest.php [hunk 7]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+namespace lewiscowles\Rfc;
```
Comment (Lewiscowles1986) [inline tests/BasicTest.php:3]: See comments on test namespacing

**Review by Lewiscowles1986 — APPROVED** [review]
Nothing seriously wrong, merge approved

File: composer.json (hunk 1 above)
Comment (peter279k) [reply composer.json:11]: According to the PHPUnit [website](https://phpunit.de/), the ```php-7.2+``` needs the PHPUnit ``6+```.

File: composer.json (hunk 2 above)
Comment (peter279k) [reply composer.json:30]: I will separate the different namespace in the next PR.

File: src/Attachment.php (hunk 3 above)
Comment (peter279k) [reply src/Attachment.php:10]: According to the PHP sample code in official PHP [website](http://php.net/manual/en/language.oop5.final.php), the keyword is ```final``` .

File: src/Envelope.php (hunk 4 above)
Comment (peter279k) [reply src/Envelope.php:8]: It's the same explanation in the previous comment.

File: tests/BasicTest.php (hunk 7 above)
Comment (peter279k) [reply tests/BasicTest.php:3]: I will do this for the next PR.

File: src/FormInput.php (hunk 6 above)
Comment (peter279k) [reply src/FormInput.php:8]: It is the same as the previous comment.

File: src/FormBody.php (hunk 5 above)
Comment (peter279k) [reply src/FormBody.php:12]: It is the same as the previous comment.

File: src/Attachment.php (hunk 3 above)
Comment (Lewiscowles1986) [reply src/Attachment.php:10]: it was running without error peter, again it was needless specificity it fixed nothing

File: composer.json (hunk 1 above)
Comment (Lewiscowles1986) [reply composer.json:11]: Interesting, thanks

**Pull request merged**